### DISCUSSION TITLE
How to extract window properties using ifcopenshell from an IFC file

### DISCUSSION THREAD

**Discussion by talha-siddiqui-7** [IfcOpenShell/IfcOpenShell#2763] [Q&A]
Hello guys,

I am trying to extract the window properties from an IFC file but having difficulties with it.
What i want to extract is [Pset_WindowCommon], [Pset_DoorWindowGlazingType]
Can someone please help me with the code? i tried using below code but its giving 'IndexError: list index out of range,

`import ifcopenshell.util.element

filename = "D:/EMJMD-SMACCs/Thesis/Thesis work/Data/SmartHome_With U values.ifc"
model = ifcopenshell.open(filename)

window = model.by_type("IfcWindow")[0]
window_type = ifcopenshell.util.element.get_type(window)
print(f"The window type of {window.Name} is {window_type.Name}")`

[IFC BIM File.zip](https://github.com/IfcOpenShell/IfcOpenShell/files/10716030/IFC.BIM.File.zip)

**Comment by aothms** [comment]
> IndexError: list index out of range

> model.by_type("IfcWindow")[0]

How sure are you the model has windows? That's the only list indexing I see happening.

**Reply by talha-siddiqui-7** [reply]
yes. i have opened the model in revit and it has windows. just dont know how to extract info about it

**Reply by aothms** [reply]
Well, open it with a text editor and search for IFCWINDOW, they are not there. Or open it in e.g BlenderBIM / Solibri / BimVision.

**Reply by talha-siddiqui-7** [reply]
i checked. you are right. this is weird. what should i do now? because i know the windows are there

**Reply by aothms** [reply]
Probably some export settings, but somebody else has to help with that.

**Reply by Moult** [reply]
I can also confirm there are no windows in the IFC. There must be an export issue in Revit.

**Reply by talha-siddiqui-7** [reply]
one thing is that i cannot view the windows and doors in 3d model or plan view. However, under families, i can clearly see windows and doors and their specifications. I have tried search for solutions related to this but cannot yet understand

**Reply by Moult** [reply]
I'm sorry, this is not the right channel for Revit support, can you try https://github.com/Autodesk/revit-ifc/ ?

**Reply by talha-siddiqui-7** [reply]
Thanks for the response. I think this might be a revit issue. I will try posting there and hopefully something helpful turns up
